### ISSUE TITLE
Presto non-optimal timestamp column queries 

### ISSUE THREAD

**Issue by lkm** [metabase/metabase#13361]
Presto doesn't support pushing down functions to it's connectors (yet). That means that any parsing would have to happen on Presto itself instead of the remote connector/database, which when you're filtering would involve a full load of all the data into Presto (full table scan in other DBs). Metabase currently generated quieries like this for Presto:

```sql
# pushdown of date_trunc('hour', "druid"."table"."__time") doesn't happen
...
WHERE date_trunc('hour', "druid"."table"."__time")
BETWEEN date_trunc('hour', date_add('hour', -4, now())) AND date_trunc('hour', date_add('hour', -1, now())) LIMIT 2000
```

Which is something that has been brought up before (https://github.com/metabase/metabase/issues/4043, https://github.com/metabase/metabase/issues/9934 and more).

In short performance on all Metabase towards Presto involving timestamp columns will be exceedingly slow. Ideally Metabase would simply generate the aforementioned query like this:

```sql
...
WHERE "druid"."table"."__time"
BETWEEN date_trunc('hour', date_add('hour', -4, now())) AND date_trunc('hour', date_add('hour', -1, now())) LIMIT 2000
```

This could be a minutiae of code that hasn't yet reached the Presto driver, as seemingly this is what https://github.com/metabase/metabase/pull/10980 aimed to do.

**Comment by flamber** [comment]
Related to #11837 (possibly duplicate)

**Comment by camsaul** [comment]
This is the same issue as #11837, implemented by #14835

**Comment by camsaul** [comment]
Fixed by #14835
